lib/template-loader.php

```diff
@@ -81,24 +81,44 @@ function gutenberg_override_query_template( $template, $type, array $templates =
  * @param array $block The root block to start traversing from.
  */
 function create_auto_draft_for_template_part_block( $block ) {
```

Comment:
Would be cool to avoid the requirement of having a "Post Id" to render a template/post/template part but I believe that's a big refactoring.

Reply:
How would you do it?

Reply:
I don't know. Maybe create temporary ids for new entities and swap them with the real ids upon save.

Reply:
Aren't `auto-drafts` temporary entities?

Reply:
yes but we have to create them, so backend logic.  What if we didn't need to create them at all and just have the first "save" create them.

Reply:
Anyway, that's not a requirement at all here, more a random thought.

Reply:
Yes, it's possible, it would just require a lot more changes and code paths in `core-data`.